marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/custom/addSourceNameAndType/main.sjs

```diff
@@ -0,0 +1,65 @@
+function main(content, options) {
+  if(options["sourceNameToAdd"] || options["sourceTypeToAdd"]) {
+    // See https://docs.marklogic.com/Node.nodeType
+    if(content.value.root && 1 === content.value.root.nodeType) {
+      addSourcesToXmlDocuments(content, options["sourceNameToAdd"], options["sourceTypeToAdd"]);
```

Comment:
Call both of these functions "addSourceTo...", since only one source is being added (even though that may require adding a "sources" property).